Assets/Scripts/RoadControl.cs

```diff
@@ -355,125 +591,467 @@ void Update()
-                        if (bd.offsetX < 0) { roadWidthOffset = screenHalfWidth * -1; }
-                        else { roadWidthOffset = screenHalfWidth; }
+                                float spriteZ;
+                                float spriteElevation = nearEdgeHeight + bb.spriteType.rect.height * nearEdgeWidthScale / 2;
+                                if (nearEdgeHeight < absHighestScreenLineDrawn && roadSegIndexOfHighestScreenLine < i) { spriteZ = spriteBgdZ; }
+                                else { spriteZ = -1f; }
 
-                        float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
+                                bb.spriteTransform.localScale = new Vector3(sizeCorrection, sizeCorrection, 1);
+                                bb.spriteTransform.position = new Vector3(x + (bb.offsetX) * nearEdgeWidthScale,
+                                                                            spriteElevation, spriteZ);
+                                bb.spriteRend.enabled = true;
+                            }
+                        }
+                    }
 
-                        bd.droid.localScale = new Vector3(sizeCorrection, sizeCorrection, 1);
-                        bd.droid.position = new Vector3(x + (bd.offsetX) * nearEdgeWidthScale,
-                                                                    nearEdgeHeight + (droidHeight + bd.droidSprite.sprite.rect.height) * nearEdgeWidthScale / 2, -1f);
-                        bd.droidSprite.enabled = true;
+                    if (visibleDroids.Count > 0) {
+                        foreach (BombDroid bd in visibleDroids) {
+
+                            if (bd.segmentIndex == i) {
+
+                                float roadWidthOffset;
+                                if (bd.offsetX < 0) { roadWidthOffset = screenHalfWidth * -1; }
+                                else { roadWidthOffset = screenHalfWidth; }
+
+                                float sizeCorrection = nearEdgeWidthScale * billboardSizeCorrectionFactor;
+
+                                bd.droid.localScale = new Vector3(sizeCorrection, sizeCorrection, 1);
+                                bd.droid.position = new Vector3(x + (bd.offsetX) * nearEdgeWidthScale,
+                                                                            nearEdgeHeight + (droidHeight + bd.droidSprite.sprite.rect.height) * nearEdgeWidthScale / 2, -1f);
+                                bd.droidSprite.enabled = true;
+                            }
+
+                            //if ((Time.time - bd.dropDelayRefTime > delayBeforeDropping) && bd.droidSprite.sprite == droidHolding) {
+                            if ((bd.segmentIndex < curSegmentIndex + 200) && bd.droidSprite.sprite == droidHolding) {
+                                bd.droidSprite.sprite = droidEmptyHanded;
+                                Bomb bmb = new Bomb();
+                                GameObject obj = new GameObject();
```

Comment:
Okay, I'm book marking this area because a lot of big data `new` allocations are happening (GameObject instances need to be very carefully tracked).
It's possible there might even be a memory leak if the objects aren't destroyed properly.

I am going to need to look at this when I have some more time, but my instinct is to remove as many unnecessary allocations and look into creating a reusable `object pool` if necessary.